### ISSUE TITLE
Rename and clarify clocked discretized continuous-time partition

### ISSUE THREAD

**Issue by HansOlsson** [modelica/ModelicaSpecification#3091]
First the name:

- It is really long, 
- The contrast is between 'clocked discretized continuous-time' and 'clocked discrete-time'; that's not very distinct
- The 'continuous-time' part is misleading; I assume the original idea was to discretize (and sometimes invert) purely continuous models and then it was extended to also discretize models with some discrete-time parts. Logically that leads to handling of models with only discrete-time parts (technically discrete-time is a sub-set of continuous-time) - but in that case 'continuous-time equations' just means 'any equations' (well, excluding initial equations).

I would thus propose to clarify and rename it to just 'clocked discretized', and instead of 'clocked discrete-time' use 'standard clocked'; or something similar (the latter is less clear). Similarly remove most uses of 'continuous-time' in its description, and make it clear that it applies to all (non-initial) equations. Obviously we can note that if the equations are not purely continuous and continuous-time _inverting_ them is problematic, but multi-rate simulation is still possible.

Then the details:
(In addition to the changes above.) There are a number of minor issues that don't work well currently related to this - they are in a separate PR https://github.com/modelica/ModelicaSpecification/pull/3090 - except for one issue:

The partitioning uses 'unknowns' and 'variables that are neither parameter nor constant' as synonyms; but in some uses, e.g., FMUs we also have top-level inputs that are kind of 'knowns' - and thus not 'unknown'. Since the structuring in sub-models should ideally not matter it makes the most sense that a top-level input are treated similarly as component-level inputs, and either are on a clock, or non-clocked; and to determine that we need to include them as variables in the clock partitioning. 

TL; DR: 'Unknowns' should clearly include 'inputs' (that are neither parameter nor constant); and possibly be renamed from 'unknowns' to be clearer.
The proposed handling of `time` should not be changed due to this.

Note that clearly stating that the 'clocked discretized partition' may be without continuous-time _states_ opens up the possibility to have a new solverMethod without specified discretization  - to handle that specific case (in particular useful if you want to ensure that there you don't accidentally have states). That's a separate less important enhancement that I ignore.

**Comment by henrikt-ma** [comment]
Sounds like a good idea to rename.

**Comment by HansOlsson** [comment]
- clocked discretized (instead of clocked discretized continuous-time; need solverMethod)
- normal clocked (instead of clocked discrete-time; no need for solverMethod); natively clocked
Try to have 'clocked' at the same spot, 'discretized clocked'?

Should think of better names (just two), but clear that change would be good.

**Comment by HansOlsson** [comment]
As I understand from my notes from the phone meeting the conclusion was to use:
- Discretized (replacing clocked discretized continuous-time)
- Clocked (replacing clocked discrete-time).
